Hardware/Hardware/CardReader/StripeCardReader/PaymentIntentParameters+Stripe.swift

```diff
@@ -4,49 +4,75 @@ import StripeTerminal
-            let applicationFeeForStripe = NSNumber(value: prepareAmountForStripe(applicationFee))
+            /// Stripe allows the credit card statement descriptor to be nil, but not an empty string
+            /// https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html#/c:objc(cs)SCPPaymentIntentParameters(py)statementDescriptor
+            builder.setStatementDescriptor(nil)
 
-            returnValue.applicationFeeAmount = applicationFeeForStripe
-        }
+            let descriptor = statementDescription ?? ""
+            if !descriptor.isEmpty {
+                builder.setStatementDescriptor(descriptor)
+            }
 
-        /// Stripe allows the credit card statement descriptor to be nil, but not an empty string
-        /// https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html#/c:objc(cs)SCPPaymentIntentParameters(py)statementDescriptor
-        returnValue.statementDescriptor = nil
-        let descriptor = statementDescription ?? ""
-        if !descriptor.isEmpty {
-            returnValue.statementDescriptor = descriptor
-        }
+            if let applicationFee = applicationFee {
+                /// Stripe requires that "The amount must be provided as a boxed UInt in the currency's smallest unit."
+                /// Smallest-unit and UInt conversion is done in the same way as for the total amount, but that does not need to be boxed.
+                let applicationFeeForStripe = NSNumber(value: prepareAmountForStripe(applicationFee))
+                builder.setApplicationFeeAmount(applicationFeeForStripe)
+            }
 
-        returnValue.receiptEmail = receiptEmail
-        returnValue.metadata = metadata
+            builder.setReceiptEmail(receiptEmail)
 
-        return returnValue
+            let updatedMetadata = prepareMetadataForStripe(with: cardReaderMetadata)
+            builder.setMetadata(updatedMetadata)
+
+            // Return payment intent built configuration:
+            return try builder.build()
     }
-}
 
-private extension Hardware.PaymentIntentParameters {
-    func prepareAmountForStripe(_ amount: Decimal) -> UInt {
-        let amountInSmallestUnit = amount * stripeSmallestCurrencyUnitMultiplier
-        return NSDecimalNumber(decimal: amountInSmallestUnit).uintValue
+    func prepareMetadataForStripe(with meta: CardReaderMetadata? = nil) -> [String: String]? {
+        let cardReaderMetadata: [String: String]? = [
+            "reader_ID": "\(meta?.readerIDMetadataKey ?? "")",
+            "reader_model": "\(meta?.readerModelMetadataKey ?? "")",
+            "platform": "\(meta?.platformMetadataKey ?? "")"
+        ]
+
+        // Updates the existing PaymentIntentParameters metadata with our cardReaderMetaData when there's a key collision in the dictionary
+        let updatedMetadata = (metadata ?? [:]).merging(cardReaderMetadata ?? [:]) { (_, new) in new }
```

Comment:
This is all pretty tricky to read. Would it be better to unwrap `meta` at the top, and return `cardReaderMetadata` merged with an empty dictionary of reader info if `meta` is nil? I _think_that's a correct reading of the code, but I'm not totally sure.

Reply:
Thanks! By unwrapping `meta` earlier this gets much simpler: dbd5a2f16a309af83d9aae1833bce55aaa5d042d

> return cardReaderMetadata merged with an empty dictionary of reader info if meta is nil?

I missed a detail here: Since metadata is of type `[String: String]?` this could should return nil if doesn't exist, not an empty dictionary. While we shouldn't expect this to happen, returning earlier fixes this and I've added an additional test case to cover this scenario: 3d2c7fe7f71cbca05f42fc0a0527e8121eaec187